erpnext-openmrs/src/main/java/com/ozonehis/eip/erpnext/openmrs/handlers/QuotationHandler.java

```diff
@@ -57,18 +57,19 @@ public boolean quotationExists(String name) {
     public Quotation getQuotation(String quotationId) {
         try (FrappeResponse response =
                 frappeClient.get("Quotation", quotationId).execute()) {
-            if (response.code() == HttpStatus.SC_OK) {
-                TypeReference<FrappeSingularDataWrapper<Quotation>> typeReference = new TypeReference<>() {};
-
-                FrappeSingularDataWrapper<Quotation> quotationWrapper = response.returnAs(typeReference);
-                return quotationWrapper.getData();
-            } else {
-                log.debug("Quotation with UUID: {} not found", quotationId);
-                return null;
+            switch (response.code()) {
+                case HttpStatus.SC_NOT_FOUND:
+                    // Quotation doesn't exist
+                    return null;
```

Comment:
Why are we not using `Optional<>` here?

Reply:
It involves more refactoring but sure let's do it.